src/Chapter3.hs

```diff
@@ -971,6 +1087,32 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data WeekDay
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday
+  deriving (Eq, Ord, Enum, Bounded, Show)
+
+isWeekend :: WeekDay -> Bool
+isWeekend weekDay
+  | weekDay > Friday = True
+  | otherwise        = False
+
+nextDay :: WeekDay -> WeekDay
+nextDay = succ
```

Comment:
Unfortunatelly, this solution won't work on Sunday, as `succ` doesn't cycle on the type 🙂 So, it would fail for `Sunday`